server/src/com/cloud/template/TemplateAdapterBase.java

```diff
@@ -325,7 +325,7 @@ public TemplateProfile prepare(RegisterIsoCmd cmd) throws ResourceAllocationExce
         Long zoneId = cmd.getZoneId();
         // ignore passed zoneId if we are using region wide image store
         List<ImageStoreVO> stores = _imgStoreDao.findRegionImageStores();
-        if (!(stores != null && stores.size() > 0)) {
+        if (!(stores != null && stores.size() > 0) && zoneId != null && zoneId > 0L) {
```

Comment:
@nvazquez would you mind changing `!(stores != null && stores.size()` to `CollectionUtils.isEmpty`?
That clause is quite confusing, and at the end of the day, it is checking if the list `stores ` is empty.

Then, you can have the following:
`if (CollectionUtils.isEmpty(stores) && zoneId != null && zoneId > 0L)`

It makes the IF condition easier to read.

Reply:
Oh sure, let me refactor it, I was just wondering how to use CollectionUtils for the zoneId but missed out the list :)